### PR TITLE
feat(doc-handling): Some investigations to use documents in the REST …

### PR DESCRIPTION
**Pull request by johnBgood**
Add support for documents in the REST Connector.

We can now access a document from the store by its id, and send it using this connector.
We can't put document in the store for now (will be done soon).

### CODE REVIEW

File: connector-sdk/core/src/main/java/io/camunda/connector/api/document/BasicDocument.java [hunk 1]
```diff
@@ -16,94 +16,40 @@
...
-    if (this == o) {
-      return true;
-    }
-    if (o == null || getClass() != o.getClass()) {
-      return false;
-    }
-    BasicDocument that = (BasicDocument) o;
-    return Objects.equals(metadata, that.metadata) && Objects.equals(content, that.content);
+  public String loadAsBase64(DocumentStore store) {
+    return new String(store.load((String) metadata.get("docRef")));
```
Comment (johnBgood) [inline connector-sdk/core/src/main/java/io/camunda/connector/api/document/BasicDocument.java:39]: we might want a `getRef()` in `Document`

File: connector-sdk/core/src/main/java/io/camunda/connector/api/document/DocumentStore.java [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+  byte[] load(String documentReference);
+
+  String store(byte[] content);
+
+  class InMemoryDocumentStore implements DocumentStore {
+    private final Map<String, byte[]> store;
+
+    public InMemoryDocumentStore() {
+      this.store = new HashMap<>();
+      store.put("example", "example".getBytes());
```
Comment (johnBgood) [inline connector-sdk/core/src/main/java/io/camunda/connector/api/document/DocumentStore.java:25]: for tests

File: connectors/http/http-base/src/main/java/io/camunda/connector/http/base/client/apache/builder/parts/ApacheRequestBodyBuilder.java [hunk 3]
```diff
@@ -126,9 +134,35 @@ private HttpEntity createMultiPartEntity(Map<?, ?> body, ContentType contentType
...
+        streamDocumentContent(entry, document, builder);
+      } else {
+        builder.addTextBody(
+            String.valueOf(entry.getKey()), String.valueOf(entry.getValue()), MULTIPART_FORM_DATA);
+      }
     }
     return builder.build();
   }
+
+  private URL getDocumentUrl(Document document) {
```
Comment (johnBgood) [inline connectors/http/http-base/src/main/java/io/camunda/connector/http/base/client/apache/builder/parts/ApacheRequestBodyBuilder.java:147]: should be in Document imo

File: connectors/http/http-base/src/main/java/io/camunda/connector/http/base/utils/DocumentHelper.java [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+package io.camunda.connector.http.base.utils;
+
+import io.camunda.document.CamundaDocument;
+import java.util.AbstractMap;
+import java.util.Collection;
+import java.util.Map;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+public class DocumentHelper {
```
Comment (johnBgood) [inline connectors/http/http-base/src/main/java/io/camunda/connector/http/base/utils/DocumentHelper.java:26]: this class will/should be improved, as it'll be used by the Webhook connector as well

File: connectors/http/http-base/src/main/java/io/camunda/connector/http/base/utils/DocumentHelper.java [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+
+  /**
+   * Traverse the {@link Map} recursively and create all Documents found in the map.
+   *
+   * @param input the input map
+   * @param transformer the transformer to apply to each document (e.g. convert to Base64 etc)
+   */
+  public Object createDocuments(Object input, Function<CamundaDocument, Object> transformer) {
```
Comment (chillleader) [inline connectors/http/http-base/src/main/java/io/camunda/connector/http/base/utils/DocumentHelper.java:34]: nit: perhaps we can rename this method to `parseDocumentsInBody` or something like that, to make it more clear what happens inside

**Review by chillleader — APPROVED** [review]
Great job 🙌

**Pull request merged**